apps/foundation-demo/src/app/checkout/pages/checkout-view/checkout-view.component.html

```diff
@@ -0,0 +1,5 @@
+<div class="checkout">
+  <div class="checkout__shipping" shipping-container #ShippingContainer="ShippingContainer">
+    <shipping-async-wrapper [isShippingInfoValid]="ShippingContainer.isShippingInfoValid$ | async"></shipping-async-wrapper>
```

Comment:
Refactor this to use transclusion, currently there's a lot of dependency between the `shipping-async-wrapper` and its components. I think something like this makes more sense.

```
<div class="checkout">
    <shipping shipping-container #ShippingContainer="ShippingContainer" 
          [isShippingInfoValid]="ShippingContainer.isShippingInfoValid$ | async" >
        <shipping-form [input]=""></shipping-form>
        <shipping-summary [input]=""></shipping-summary>
    </shipping>
</div>
```

Reply:
When I looked into this more deeply, I found that I couldn't use transclusion here. The reason is, I need to use `isShippingInfoValid` on shipping-form and shipping-summary, and I can't use it on them if it is still async. So I need the shipping component to process the async isShippingInfoValid$. Though I did restrict access to ShippingContainer to this component only, so I'm passing all references that come from ShippingContainer through the shipping component